### PR TITLE
Create presto-util module for utilities used by presto-main and connectors

### PR DESCRIPTION
**Pull request by findepi**
`presto-util` module is meant to hold utilities used by presto-main and, connectors distributed with Presto.

### CODE REVIEW

**Comment by electrum** [comment]
`filteredCopy` is a method I've long wanted, but we've never had a good place to put it, and by itself it never had enough "weight" to justify a new module.

**Comment by electrum** [comment]
I merged the `QueryExplanation` commit

**Comment by findepi** [comment]
I named it util just as the `c.f.presto.util` package is named, but i agree it might become junkish. However, operations on collections is not the only thing I would like to have shared. #8495 is another candidate. @electrum how do you see it going forward?

File: pom.xml [hunk 1]
```diff
@@ -1032,7 +1032,7 @@
                                         </goals>
                                     </pluginExecutionFilter>
                                     <action>
-                                        <ignore />
+                                        <ignore/>
```
Comment (electrum) [inline pom.xml]: The original formatting is correct. There should be a space before the `/>` for a self-closed XML tag. I'll update https://github.com/airlift/codestyle to match

**Comment by electrum** [comment]
The locking should go in Airlift concurrent (I commented there).

Going forward, we'll stick with the same approach of either using singularly focused libraries or copy/paste of utility code. Shared libraries should have a higher bar in terms of API design, unit tests, etc. It's a definite trade-off and matter of opinion. We prefer to err on the side of copy/pasted utility methods than having a shared junk drawer.

**Comment by electrum** [comment]
It's generally easier to cleanup copy/pasted utility methods than an it is to cleanup a shared junk drawer that has lots of random users. Here's an example of the latter that was a mistake: https://github.com/airlift/airlift/pull/540

File: pom.xml (hunk 1 above)
Comment (electrum) [inline pom.xml]: Update: https://github.com/airlift/codestyle/pull/5

**Comment by findepi** [comment]
Thanks for the XML formatting fixes. I'll update my IDE cfg when this is merged.

* locks - i made a PR in airlift
* the classes i move here - `presto-collections` is a fine name for them

i have one more example of functionality that I'd like to have shared. Currently we have 3 copies of classes `AbstractAggregatedMemoryContext`, `AggregatedMemoryContext`, `LocalMemoryContext`. I'd like to improve their API (method `updateBytes` has vague meaning). Since this looks like useful for connectors (already used in parquet and ORC readers to satisfy SPI requirements) I'd prefer to have it shared rather than manually kept in sync.

If this is in presto and meant for internal use only (presto main and those connectors we maintain), potential cleanup shouldn't be harder than when we have copies. @electrum what do you think? If i didn't convince you, I'll rename to module to `-collections` (or `-collect`?)

**Comment by electrum** [comment]
I think `AbstractAggregatedMemoryContext` is a temporary thing. @haozhun knows more about that. Having a couple copies of a trivial class is definitely not worth creating a junk drawer.

**Pull request closed without merging**